pkg/cmd/cli/cmd/startbuild.go

```diff
@@ -58,11 +64,28 @@ Examples:
 				newBuild = buildutil.GenerateBuildFromBuild(build)
 			}
 
+			var wg sync.WaitGroup
+			if streamLogs {
+				rd, err := client.Get().Namespace(namespace).Prefix("redirect").Resource("buildLogs").Name(buildName).Stream()
+				checkErr(err)
+				wg.Add(1)
+				go func(rd io.ReadCloser, out io.Writer) {
+					defer wg.Done()
+					defer rd.Close()
+
+					_, err = io.Copy(out, rd)
+					checkErr(err)
+				}(rd, out)
+
+			}
+
 			newBuild, err = client.Builds(namespace).Create(newBuild)
 			checkErr(err)
+			wg.Wait()
 			fmt.Fprintf(out, "%s\n", newBuild.Name)
 		},
 	}
 	cmd.Flags().String("from-build", "", "Specify the name of a build which should be re-run")
+	cmd.Flags().Bool("stream-logs", false, "Run a build and stream its logs")
```

Comment:
Users will learn about follow from the help, and it's reasonably consistent with other command line tools.  The logging aspect is just one part of it - you're also watching to see the state of the build.  In the long run, the log streaming needs to be much more sophisticated at the apiserver level so that we can chain logs from multiple actions (build -> deployment process pod).

> On Mar 2, 2015, at 5:28 PM, Michail Kargakis <notifications@github.com> wrote:
> 
> In pkg/cmd/cli/cmd/startbuild.go:
> 
> >  			fmt.Fprintf(out, "%s\n", newBuild.Name)
> >  		},
> >  	}
> >  	cmd.Flags().String("from-build", "", "Specify the name of a build which should be re-run")
> > +	cmd.Flags().Bool("stream-logs", false, "Run a build and stream its logs")
> From a user pov it's not clear what --follow is doing so if there is space for argument here I would still root for --stream-logs.
> 
> —
> Reply to this email directly or view it on GitHub.
>